sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/CachedDekProperties.java

```diff
@@ -13,7 +13,7 @@ public CachedDekProperties(
         DataEncryptionKeyProperties serverProperties,
         Instant serverPropertiesExpiryUtc) {
         assert(serverProperties != null);
-
+        // TODO: add NPE validation
```

Comment:
NIT: wouldn't it make sense to do that in this PR vs. just adding TODO? Or is the work more involved than I naively assume?

Reply:
TODO comment is not accurate. will update the TODO to capture the details. 

on contentResponseOnWriteEnabled = false we get into NPE.
I need to think on what to do on that case (for contentResponseOnWriteEnabled =false). 
Just NPE validation won't help here.

Reply:
Makes sense